### PR TITLE
allow no blueprints to load

### PR DESCRIPTION
**Pull request by onufer**
## Description

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/reactor/reactors.py [hunk 1]
```diff
@@ -2257,30 +2257,32 @@ def processLoading(self, cs, dbLoad: bool = False):
...
-                runLog.header(
-                    "=========== Axially expanding blueprints assemblies (except control) from Tinput to Thot ==========="
-                )
-                self._applyThermalExpansion(
-                    self.parent.blueprints.assemblies.values(),
-                    dbLoad,
-                    finestMeshAssembly,
-                )
+            if not self.parent.blueprints:
+                # we don't require reactors to have blueprints
```
Comment (onufer) [inline armi/reactor/reactors.py:2261]: we dont require BP, so we should allow load without them
https://github.com/terrapower/armi/blob/cacbd3d30d58e713348328d3a745cbe4adfdcc65/armi/bookkeeping/db/database3.py#L832-L836

**Review by opotowsky — CHANGES_REQUESTED** [review]
Simple change, and looks good to me! But we have a testing problem (that isn't your fault)

The reason for the code coverage reduction is that the code here wasn't originally covered by a test, and this PR adds lines to that uncovered `else`: 
``` python
            else:
                runLog.warning(
                    f"Blueprints file not found at {bpPath}"
                    "not writing blueprints to database."
                )
                bpString = ""
```

Would you mind adding a little test that loads `TEST_ROOT`, modifies `loadingfile` to an empty string, and tests that `initDB` or `writeInputsToDB` gives `self.h5db["inputs/blueprints"] = ""`?

File: armi/reactor/reactors.py [hunk 2]
```diff
@@ -2253,34 +2253,37 @@ def processLoading(self, cs, dbLoad: bool = False):
...
-                runLog.header(
-                    "=========== Axially expanding blueprints assemblies (except control) from Tinput to Thot ==========="
-                )
-                self._applyThermalExpansion(
-                    self.parent.blueprints.assemblies.values(),
-                    dbLoad,
-                    finestMeshAssembly,
-                )
+            # we don't require reactors to have blueprints
+            if self.parent.blueprints:
```
Comment (jakehader) [inline armi/reactor/reactors.py:2257]: I think we _should_ require this going forward though. Databases need to have blueprints on them for `createAssemblyOfType` to function when loading from a database. Otherwise, we cannot do things like load from a state point, swap a fresh assembly, run a calculation. Is the plan to add this elsewhere?

Comment (jakehader) [reply armi/reactor/reactors.py:2257]: Spoke offline and we agreed that BP in the database will be enforced later.

**Comment by onufer** [comment]
Abandoning because there are other places blueprints are addressed. we really need them.

**Comment by john-science** [comment]
> Abandoning because there are other places blueprints are addressed. we really need them.

Also, blueprints can be addressed (and then hypothetically changed) at almost any time in ARMI or any plugin that imports ARMI.

I have tried to make it harder to do that, but it is a _very_ common idea to write your code around changing the blueprints object. So, fighting blueprint modification is always a lot of work.
